src/game.h

```diff
@@ -443,8 +443,8 @@ class game
 
         void interactive_inv();
 
-        int inv_for_filter( const std::string &title, item_filter filter, const std::string &none_message = "" );
-        int inv_for_filter( const std::string &title, item_location_filter filter, const std::string &none_message = "" );
+        int inv_for_filter( const std::string &title, const item_filter &filter, const std::string &none_message = "" );
```

Comment:
As @Coolthulhu pointed out, the majority of consumers would want item. I'd spare them the conversion and leave it as it is.